File: php/src/Model/BankAccount.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Model;

use Dokmatiq\DocGen\Internal\Serializer;

/** Bank account details for payment. */
final class BankAccount
{
    public function __construct(
        public readonly string $iban,
        public readonly ?string $bic = null,
        public readonly ?string $accountHolder = null,
    ) {}

    /** @return array<string, mixed> */
    public function toArray(): array
    {
        return Serializer::toArray($this);
    }
}

File: java/src/main/java/com/dokmatiq/docgen/model/BankAccount.java
package com.dokmatiq.docgen.model;

import com.fasterxml.jackson.annotation.JsonInclude;

/** Bank account details for payment. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record BankAccount(
        String iban,
        String bic,
        String accountHolder
) {
    public BankAccount(String iban) {
        this(iban, null, null);
    }
}
